### ISSUE TITLE
Class 'Error' not found & PHP bug #42098/#54054/#65322

### ISSUE THREAD

**Issue by Savageman** [fuel/core#1617]
I just posted a comment about the commit supposed to fix this PHP bug.

I have very easy step to reproduce, just add the following code at the end of `app/bootstrap.php`:

``` php
function buggyCall($arg) {
    $arg = 'bar';
}

$arg = 'foo';
buggyCall(&$arg);
```

You should see "[ Fatal Error ]: Class 'Log' not found".

**Comment by WanWizard** [comment]
I don't, I get a standard PHP (xdebug) error? 

It is a parsing error, so the error is thrown by PHP before the code is executed. And since the app bootstrap loads the core bootstrap which activates the error handler, there is no error handler when this error occurs. If I add a runtime error (call Unknown::unknown() for example), the error handler kicks in without problems.

**Comment by WanWizard** [comment]
If I move it to somewhere else (I tried app/config/config.php), the error handler kicks in without problems, and shows me a "Compile Error".

**Comment by Savageman** [comment]
Oh right sorry, I forgot. This error is on PHP 5.3.

**Comment by WanWizard** [comment]
That problem will fix itself soon, it will be EOL in a few months ;)

**Comment by Savageman** [comment]
So Fuel doesn't support 5.3? ;)

I could try to find some others E_Strict or E_Deprecated error which triggers the issue in 5.4 or 5.5, I think it's possible (the PHP bugfix is only available for 5.6).

**Comment by Savageman** [comment]
This is my "Fix" for reference:
https://github.com/fuel/core/commit/87fdfae80a779a2657671942672a79432be5f678#commitcomment-5142980

``` php
// Autoloader not available (happens when an error occurs at compile time), fallback nicely
// https://bugs.php.net/bug.php?id=65322
// The error will be logged by native PHP instead of custom Fuel error file
if (!class_exists('Error')) {
    return false;
}
```

**Comment by WanWizard** [comment]
Fuel v1 is currently still 5.3. v2 will be either 5.4 or 5.5, we haven't decided yet. ;)

My problem still is that if I can't reproduce it, I can't fix it. No matter what I try, Fuel's error handler kicks in without problems. I'll try to setup a 5.3 test environment to see if I can reproduce it there.

**Comment by joelvanpatten** [comment]
It seems like it might be related to [this error](https://github.com/kenjis/AspectMock-Strange-Error), in which case it appears to happen in certain OS/Php combinations but not others.

**Comment by joelvanpatten** [comment]
If it is the same issue, I can certainly reproduce it in CentOS 6.5 using either PHP 5.5 or PHP 5.6.  If they are not related, then I apologize for wasting your time (although I suppose its worth mentioning that the linked issue is certainly affecting me I have spent the past two days trying to find a solution, but so far haven't found one).

**Comment by WanWizard** [comment]
We run everything on CentOS, most still 6.5, new ones on 7. With PHP from the Remi repo's. And I've never seen this issue.

In previous responses it was limited to 5.3, and a fix for 5.6. But if you have it too, it's clearly not fixed in 5.6. 

You you probably have to resort to Savageman's hack to solve your problem.

**Comment by kenjis** [comment]
@Savageman 

My fix about it.

``` diff
--- a/fuel/app/bootstrap_phpunit.php
+++ b/fuel/app/bootstrap_phpunit.php
@@ -66,3 +66,6 @@ Fuel::$is_test = true;
// Import the TestCase class
import('testcase');
+
+// Ad hoc fix for AspectMock error
+class_exists('Error');
```
